### PR TITLE
Fix a bug in the getLocationFromAbsolutePath() method

### PR DESCRIPTION
**Pull request by deepakwadhwani**
Fix a bug in the getLocationFromAbsolutePath() method
Use getHomeLocation  instead

Build running : http://builds.cask.co/browse/CDAP-DUT5328-21

### CODE REVIEW

**Comment by chtyim** [comment]
Can you add a `LocationsTest` for that?

**Comment by chtyim** [comment]
Note that we need to test location home uri in `maprfs:///path` format as well as the HDFS format `hdfs://host/path`

File: cdap-common/src/main/java/co/cask/cdap/common/io/Locations.java [hunk 1]
```diff
@@ -344,15 +344,19 @@ public static Location getParent(Location location) {
   /**
    * Given a locationPath which is derived from URI.getPath() and a locationFactory
    * this method generates a Location with URI generated from the given Absolute path
+   * The given absolutePath is expected to contain basePath if any.
```
Comment (albertshau) [inline cdap-common/src/main/java/co/cask/cdap/common/io/Locations.java]: this entire javadoc is confusing and should change. Just say something like 'Get a Location using a specified location factory and absolute path.'

File: cdap-common/src/main/java/co/cask/cdap/common/io/Locations.java [hunk 2]
```diff
@@ -344,15 +344,19 @@ public static Location getParent(Location location) {
...
    * @return Location resulting from absolute locationPath
    */
   public static Location getLocationFromAbsolutePath(LocationFactory locationFactory, String absolutePath) {
-    URI basePathURI = URI.create(locationFactory.create("").toURI().getPath());
-    URI locationURI = URI.create(absolutePath);
-    URI relativePathURI = basePathURI.relativize(locationURI);
-    return locationFactory.create(relativePathURI);
+    URI homeURI = locationFactory.getHomeLocation().toURI();
+    try {
+      return locationFactory.create(new URI(homeURI.getScheme(), homeURI.getAuthority(), absolutePath, null));
```
Comment (anwar6953) [inline cdap-common/src/main/java/co/cask/cdap/common/io/Locations.java]: See a Twill PR here: https://github.com/apache/twill/pull/28
Would a similar problem be here?

**Review by chtyim — COMMENTED** [review]
LGTM

**Comment by deepakwadhwani** [comment]
Clean Build : http://builds.cask.co/browse/CDAP-DUT5328-22

**Pull request merged**